server.h

```diff
@@ -0,0 +1,240 @@
+                                     if (accessor.Rows().has(device_id)) {
+                                       assert(accessor[device_id].size() == 1);
+                                       uid = accessor[device_id].begin()->uid;
+                                     }
+                                   }).Go();
+
+              if (uid != UID::INVALID) {  // Existing user.
+                // Invalidating all old tokens.
+                storage_.Transaction([&](StorageAPI::T_DATA data) {
+                                       auto mutator = MatrixEntry<UIDTokenPair>::Mutator(data);
+                                       for (const auto& uid_token : mutator[uid]) {
+                                         UIDTokenPair invalid_token(uid_token.uid, uid_token.token, false);
+                                         mutator.Add(invalid_token);
+                                       }
+                                       user = data.Get(uid);
+                                     }).Go();
+                user_entry.score = user.score;
+              } else {  // New user.
+                uid = RandomUID();
+                user.uid = uid;
+                storage_.Add(user).Go();
+              }
+
+              CopyUserInfoToResponseEntry(user, user_entry);
+              // Generating a new token.
+              const std::string token = RandomToken();
+              user_entry.token = token;
+              storage_.Add(UIDTokenPair(uid, token)).Go();
+
+              RespondWithFeed(user_entry, FromString<size_t>(r.url.query.get("feed_count", "20")),
+                              std::move(r));
+            }
+          }
+        });
+
+    HTTP(http_port_)
+        .Register("/feed", [this](Request r) {
+          const UID uid = StringToUID(r.url.query["uid"]);
+          const std::string token = r.url.query["token"];
+          if (r.method != "GET") {
+            r("METHOD NOT ALLOWED\n", HTTPResponseCode.MethodNotAllowed);
+          } else {
+            if (uid == UID::INVALID) {
+              r("NEED VALID UID\n", HTTPResponseCode.BadRequest);
+            } else {
+              bool valid_token = storage_.Get(uid, token).Go();
+              if (!valid_token) {
+                r("NEED VALID TOKEN\n", HTTPResponseCode.Unauthorized);
+              } else {
+                const auto user = storage_.Get(uid).Go();
+                ResponseUserEntry user_entry;
+                CopyUserInfoToResponseEntry(user, user_entry);
+                user_entry.token = token;
+                RespondWithFeed(user_entry, FromString<size_t>(r.url.query.get("feed_count", "20")),
+                                std::move(r));
+              }
+            }
+          }
+        });
+  }  // CTFOServer(...)
+
+  void RespondWithFeed(ResponseUserEntry user_entry, size_t max_count, Request r) {
+    storage_.Transaction(
+        [&, user_entry, max_count](StorageAPI::T_DATA data) {
```

Comment:
I remember we chat about it, yet, once again: what does the `[&]` capture here and can it be avoided?